src/devices/GrovePi/Sensors/DhtSensor.cs

```diff
@@ -63,9 +63,16 @@ public double[] Value
         /// </summary>
         public void Read()
         {
-            _grovePi.WriteCommand(GrovePiCommand.DhtTemp, _port, (byte)DhtType, 0);
+            // Fix for the firmware 1.4.0, you can request a new measurement only if you got data
+            // from the previous one
+            if (!(double.IsNaN(_lastTemHum[0]) && double.IsNaN(_lastTemHum[1])))
+            {
+                _grovePi.WriteCommand(GrovePiCommand.DhtTemp, _port, (byte)DhtType, 0);
+            }
             // Wait a little bit to read the result
-            Thread.Sleep(50);
+            // Delay from source code, line 90 in the Grove Pi firmware repository
+            // This is needed for firmware 1.4.0 and also working for previous versions
+            Thread.Sleep(300);
```

Comment:
is this large delay always needed or only the first time when both values are NaN?

Reply:
Yes, it is a large delay. If I don't do a delay like this, you'll get multiple NaN after reading. I know most people will prefer to read a correct value even if it takes long time.
In the previous version of the firmware, the old value was returned so you always had a value. With this new version, you get a NaN if the value is not present.